flink-connector-kudu/src/main/java/org/apache/flink/connectors/kudu/table/utils/KuduTypeUtils.java

```diff
@@ -0,0 +1,152 @@
+
+import org.apache.kudu.ColumnTypeAttributes;
+import org.apache.kudu.Type;
+
+import java.sql.Timestamp;
+
+import static org.apache.flink.util.Preconditions.checkNotNull;
+
+public class KuduTypeUtils {
+
+    public static DataType toFlinkType(Type type, ColumnTypeAttributes typeAttributes) {
+        switch (type) {
+            case STRING:
+                return DataTypes.STRING();
+            case FLOAT:
+                return DataTypes.FLOAT();
+            case INT8:
+                return DataTypes.TINYINT();
+            case INT16:
+                return DataTypes.SMALLINT();
+            case INT32:
+                return DataTypes.INT();
+            case INT64:
+                return DataTypes.BIGINT();
+            case DOUBLE:
+                return DataTypes.DOUBLE();
+            case DECIMAL:
+                return DataTypes.DECIMAL(typeAttributes.getPrecision(), typeAttributes.getScale());
+            case BOOL:
+                return DataTypes.BOOLEAN();
+            case BINARY:
+                return DataTypes.BYTES();
+            case UNIXTIME_MICROS:
+                return new AtomicDataType(new TimestampType(3), Timestamp.class);
+
+            default:
+                throw new IllegalArgumentException("Illegal var type: " + type);
+        }
+    }
+
+    public static Type toKuduType(DataType dataType) {
+        checkNotNull(dataType, "type cannot be null");
+        LogicalType logicalType = dataType.getLogicalType();
+        return logicalType.accept(new KuduTypeLogicalTypeVisitor(dataType));
+    }
+
+    private static class KuduTypeLogicalTypeVisitor extends LogicalTypeDefaultVisitor<Type> {
+
+        private final DataType dataType;
+
+        KuduTypeLogicalTypeVisitor(DataType dataType) {
+            this.dataType = dataType;
+        }
+
+        @Override
+        public Type visit(BooleanType booleanType) {
+            return Type.BOOL;
+        }
+
+        @Override
+        public Type visit(TinyIntType tinyIntType) {
+            return Type.INT8;
+        }
+
+        @Override
+        public Type visit(SmallIntType smallIntType) {
+            return Type.INT16;
+        }
+
+        @Override
+        public Type visit(IntType intType) {
+            return Type.INT32;
+        }
+
+        @Override
+        public Type visit(BigIntType bigIntType) {
+            return Type.INT64;
+        }
+
+        @Override
+        public Type visit(FloatType floatType) {
+            return Type.FLOAT;
+        }
+
+        @Override
+        public Type visit(DoubleType doubleType) {
+            return Type.DOUBLE;
+        }
+
+        @Override
+        public Type visit(DecimalType decimalType) {
+            return Type.DECIMAL;
+        }
+
+        @Override
+        public Type visit(TimestampType timestampType) {
+            return Type.UNIXTIME_MICROS;
+        }
+
+        @Override
+        public Type visit(VarCharType varCharType) {
```

Comment:
Kudu 1.12.0 will have VARCHAR

Reply:
👍

Reply:
VarChar seems to be not yet supported by the Blink planner